qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,306 @@
+            (\Delta_{pq} a^\dagger_p a^\dagger_q + \text{h.c.})
+        + \text{constant}
+
+    where :math:`M` is a Hermitian matrix and :math:`\Delta` is an antisymmetric matrix.
+    """
+
+    def __init__(
+        self,
+        hermitian_part: Optional[np.ndarray] = None,
+        antisymmetric_part: Optional[np.ndarray] = None,
+        constant: float = 0.0,
+    ) -> None:
+        r"""Initialize a QuadraticHamiltonian.
+
+        Args:
+            hermitian_part: The matrix :math:`M` containing the
+                coefficients of the terms that conserve particle number.
+            antisymmetric_part: The matrix :math:`\Delta` containing the
+                coefficients of the terms that do not conserve particle number.
+            constant: An additive constant term.
+
+        Raises:
+            ValueError: Either a Hermitian part or antisymmetric part must be specified.
+        """
+        if hermitian_part is not None:
+            self._n_orbitals, _ = hermitian_part.shape
+        elif antisymmetric_part is not None:
+            self._n_orbitals, _ = antisymmetric_part.shape
+        else:
+            raise ValueError("Either a Hermitian part or antisymmetric part must be specified.")
+
+        # TODO maybe validate matrix properties
+        self._hermitian_part = hermitian_part
+        self._antisymmetric_part = antisymmetric_part
+        self._constant = constant
+
+        if self._hermitian_part is None:
+            self._hermitian_part = np.zeros((self._n_orbitals, self._n_orbitals))
+        if self._antisymmetric_part is None:
+            self._antisymmetric_part = np.zeros((self._n_orbitals, self._n_orbitals))
+
+    @property
+    def hermitian_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that conserve particle number."""
+        return self._hermitian_part
+
+    @property
+    def antisymmetric_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that do not conserve particle number."""
+        return self._antisymmetric_part
+
+    @property
+    def constant(self) -> float:
+        """The constant."""
+        return self._constant
+
+    def _fermionic_op(self) -> FermionicOp:
+        """Convert to FermionicOp."""
+        terms = []
+        # TODO I shouldn't have to use string labels
+        # diagonal terms
+        for i in range(self._n_orbitals):
+            terms.append((f"+_{i} -_{i}", self.hermitian_part[i, i]))
+        # off-diagonal terms
+        for i in range(self._n_orbitals):
+            for j in range(i + 1, self._n_orbitals):
+                terms.append((f"+_{i} -_{j}", self.hermitian_part[i, j]))
+                terms.append((f"+_{j} -_{i}", self.hermitian_part[j, i]))
+                terms.append((f"+_{i} +_{j}", self.antisymmetric_part[i, j]))
+                terms.append((f"-_{j} -_{i}", self.antisymmetric_part[i, j].conj()))
+        # constant
+        terms.append(("", self.constant))
```

Comment:
How about you initialize `terms = [("", self.constant)]` before entering the loop above?

Reply:
Done.